### PR TITLE
[RISCV][compiler-rt] create __riscv__cpu_model for vendorID, ArchID, …

### PR DESCRIPTION
**Pull request by BeMg**
…ImplID

This patch 

1. remove the vendorId from `__riscv_vendor_feature_bits`
2. Define a new structure for vendorID, ArchID and ImplID
3. Update the relate init code

### CODE REVIEW

File: compiler-rt/lib/builtins/cpu_model/riscv.c [hunk 1]
```diff
@@ -16,11 +16,16 @@ struct {
...
-  unsigned vendorID;
   unsigned length;
   unsigned long long features[RISCV_VENDOR_FEATURE_BITS_LENGTH];
 } __riscv_vendor_feature_bits __attribute__((visibility("hidden"), nocommon));
 
+struct {
+  unsigned mVendorID;
+  unsigned mArchID;
+  unsigned mImplID;
+} __riscv__cpu_model __attribute__((visibility("hidden"), nocommon));
```
Comment (kito-cheng) [inline compiler-rt/lib/builtins/cpu_model/riscv.c:27]: __riscv__cpu_model -> __riscv_cpu_model for naming consistency

**Comment by asb** [comment]
Alongside #101472, this seems to be extending the interface beyond what is described in the [current draft spec](https://github.com/riscv-non-isa/riscv-c-api-doc/pull/74). Is there some other source of truth I should be looking for? Is this a proposal to change the spec, or an implementation of something that was already decided (at least tentatively)?

**Review by kito-cheng — APPROVED** [review]
LGTM, and I believe this modification will allow us to quickly extend the `cpu=` or `tune=` syntax in the future :)

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**